src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RazorHtmlCSharpLanguageServer.cs

```diff
@@ -60,9 +59,22 @@ public Task<InitializeResult> InitializeAsync(JToken input, CancellationToken ca
             // InitializeParams only references ClientCapabilities, but the VS LSP client
             // sends additional VS specific capabilities, so directly deserialize them into the VSClientCapabilities
             // to avoid losing them.
-            _clientCapabilities = input["capabilities"].ToObject<VSClientCapabilities>();
+            var clientCapabilities = input["capabilities"].ToObject<VSClientCapabilities>();
             var initializeParams = input.ToObject<InitializeParams>();
-            return ExecuteRequestAsync<InitializeParams, InitializeResult>(Methods.InitializeName, initializeParams, _clientCapabilities, cancellationToken);
+            return ExecuteRequestAsync<InitializeParams, InitializeResult>(Methods.InitializeName, initializeParams, clientCapabilities, cancellationToken);
+        }
+
+        [JsonRpcMethod(Methods.TextDocumentCompletionName)]
+        public Task<SumType<CompletionItem[], CompletionList>?> ProvideCompletionsAsync(JToken input, CancellationToken cancellationToken)
+        {
+            if (input is null)
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            var clientCapabilities = new VSClientCapabilities();
```

Comment:
Why'd you stop using _clientCapabilities?

Reply:
Because it wasn't being used in other places and the source of it will be set differently by different handlers.

Reply:
Hmm? There's only 1 set of capabilities though. For instance the capabilities let us know what the client supports